### ISSUE TITLE
Перерасчет прямой модели под заданные координаты сенсоров ЭЭГ

### ISSUE THREAD

**Issue by logogriph2** [dmalt/cognigraph#21]
Добрый день!

Скажите, где у вас реализован перерасчет прямой модели под заданные координаты сенсоров ЭЭГ-шлема по заданному файлу корегистрации? В каком виде нудно предавать координаты сенсоров и где лежат дефолтные координаты?

**Comment by dmalt** [comment]
Да, это сделано в виде инструмента командной строки. Путь к тулу scripts/recompute_forward.py.

**Comment by logogriph2** [comment]
а можете указать расположение файла с дефолтными координатами и привести пример использования?

**Comment by dmalt** [comment]
Не очень понимаю, что такое файл с дефолтными координатами. Вообще координаты сенсоров хранятся в самом файле прямой модели. В этой утилите координаты подаются через tsv-файл,
в котором первая колонка - имена каналов, а остальные три - x,y и z этого сенсора в R^3. Можно еще пропускать первые n колонок. Если нужны координаты монтажа 1005, например, их можно вытащить [отсюда](https://drive.google.com/open?id=1D9LLvDVXYI1gF5dPpZcaYPh0TW-eyHxj). По ссылке прямая модель для монтажа 1005.
Вот пример перерасчета старой прямой модели под новое количество источников:
`python recompute_forward.py ../tests/data/sample_1005-eeg-oct-6-fwd.fif sample -s ~/mne_data/MNE-sample-data/subjects --spacing 'ico4' -d ../tests/data/sample_1005-eeg-ico-4-fwd.fif`
А вот так перерасчитывать под новые координаты сенсоров:
`python recompute_forward.py ../tests/data/dmalt_custom-fwd.fif ~/Data/cognigraph/data/sample_aligned-trans.fif sample -s ~/mne_data/MNE-sample-data/subjects -p 1 -c ch_locs.pos`
Вот пример файла с координатами сенсоров:
[ch_locs.pos.txt](https://github.com/dmalt/cognigraph/files/2468960/ch_locs.pos.txt)
нужно только убрать разрешение .txt -- гитхаб без него не хотел добавлять файл

**Comment by logogriph2** [comment]
было бы здорово включить этот скрипт в интерфейс

**Comment by logogriph2** [comment]
В скрипте recompute_forward.py парсинг аргументов записан как:
parser.add_argument('channels', type=argparse.FileType('r'),
                    help='File with tab-separated channel locations')
parser.add_argument('forward', type=argparse.FileType('r'),
                    help='Path to forward solution')
parser.add_argument('trans', type=argparse.FileType('r'),
                    help='Path to trans file')
parser.add_argument('subject_name',# required=True,
                    help='Subject name in FreeSurfer folder')
parser.add_argument('-s', '--subjects-dir', default=os.getenv('SUBJECTS_DIR'),
                    help='Path to Freesurfer`s SUBJECTS_DIR.' +
                    ' Defaults to env variable with the same name')
parser.add_argument('-d', '--dest',
                    default='{}-fwd.fif'.format(time.strftime("%d_%m_%y")),
                    help='Destination file')
parser.add_argument('-p', '--padding', type=int, default=0,
                    help='Skip p columnts from the left in channels file')
parser.add_argument('-j', '--n_jobs', type=int, default=1,
                    help='Number of jobs for forward solution computation')
parser.add_argument('-f', '--force', action='store_true',
                    help='If specified, overwrite destination file' +
                    ' without prompting')

Это не очень соотносится с вашей инструкцией.

Команда python recompute_forward.py C:\DL_projects\cognigraph-master\tests\data/ch_locs_stanford1.pos C:\DL_projects\cognigraph-master\tests\data/DF_2018-03-02_11-34-38.fif -s C:\DL_projects\cognigraph-master\tests\data -p 1 не выполняется

**Comment by dmalt** [comment]
В текущей версии recompute_forward порядок аргументов другой.
https://github.com/dmalt/cognigraph/blob/master/scripts/recompute_forward.py#L17

Обновите код, после этого все должно заработать.

**Comment by mahakomar11** [comment]
Каким должен быть второй аргумент?
parser.add_argument('subject_name',  # required=True,
                    help='Subject name in FreeSurfer folder')

**Comment by dmalt** [comment]
Второй аргумент - это имя папки, в котрой лежит анатомия. По фрисёрферовскому стандарту она как правило называется по имени испытуемого. Для тестового датасета mne-python это sample
